qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,176 @@
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run readout spectroscopy.
+            frequencies: The frequencies to scan in the experiment, in Hz.
+            backend: Optional, the backend to run the experiment on.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                resonator frequency in the backend.
+        """
+        super().__init__(qubit, frequencies, backend, absolute)
+        self.analysis = ResonatorSpectroscopyAnalysis()
+        self._set_analysis_data_processor()
+
+    def _set_analysis_data_processor(self):
+        """Keep the data processor consistent with the run options."""
+        processor = get_processor(
+            self.run_options.meas_level,
+            self.run_options.meas_return,
+            self.analysis.options.normalization,
+            dimensionality_reduction=ProjectorType.ABS,
+        )
+        self.analysis.set_options(data_processor=processor)
+
+    def set_run_options(self, **fields):
+        """Wrap set run options to keep the data processor consistent."""
+        super().set_run_options(**fields)
+        self._set_analysis_data_processor()
+
+    @property
+    def center_frequency(self) -> float:
+        """Returns the center frequency of the experiment.
+
+        Returns:
+            The center frequency of the experiment.
+
+        Raises:
+            QiskitError: If the experiment does not have a backend set.
+        """
+        if self.backend is None:
+            raise QiskitError("backend not set. Cannot call center_frequency.")
+
+        return self.backend.defaults().meas_freq_est[self.physical_qubits[0]]
+
+    def _meas_template_circuit(self) -> QuantumCircuit:
+        """Return the template quantum circuit."""
+        circuit = QuantumCircuit(1, 1)
+        circuit.measure(0, 0)
+
+        return circuit
+
+    def _spec_gate_schedule(
+        self, backend: Optional[Backend] = None
+    ) -> Tuple[pulse.ScheduleBlock, Parameter]:
+        """Create the spectroscopy schedule."""
+
+        qubit = self.physical_qubits[0]
+
+        freq_param = Parameter("frequency")
+
+        with pulse.build(backend=backend, name="spectroscopy") as schedule:
+            pulse.shift_frequency(freq_param, pulse.MeasureChannel(qubit))
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration=self.experiment_options.duration,
+                    amp=self.experiment_options.amp,
+                    sigma=self.experiment_options.sigma,
+                    width=self.experiment_options.width,
+                ),
+                pulse.MeasureChannel(qubit),
+            )
+            pulse.acquire(self.experiment_options.duration, qubit, pulse.MemorySlot(0))
+
+        return schedule, freq_param
+
+    def circuits(self):
```

Comment:
Because in resonator spectroscopy we are attaching the pulse schedule to the measurement instruction which does not have a parameter. Therefore, we can not do:
```python
circuit.add_calibration("measure", self.physical_qubits, sched_)
assigned_circ = circuit.assign_parameters({freq_param: freq_shift}, inplace=False)
```
for example, the following simple test case will raise:
```
import qiskit.pulse as pulse
from qiskit.circuit import Parameter
from qiskit import QuantumCircuit

param = Parameter("amp")

with pulse.build(backend=None) as sched:
    pulse.play(pulse.Drag(160, param, 2,0, 40), pulse.DriveChannel(2))
    
circ = QuantumCircuit(1, 1)
circ.measure(0, 0)
circ.add_calibration("measure", (2, ), sched)
circ.assign_parameters({param: 0.2345566}, inplace=False)
```